gradle-plugin/src/main/java/com/squareup/hephaestus/plugin/DisableIncrementalCompilationTask.kt

```diff
@@ -1,60 +1,43 @@
+@file:Suppress("UnstableApiUsage")
+
 package com.squareup.hephaestus.plugin
 
 import org.gradle.api.DefaultTask
-import org.gradle.api.artifacts.ArtifactView
-import org.gradle.api.artifacts.ResolvableDependencies
-import org.gradle.api.attributes.Attribute
-import org.gradle.api.file.FileCollection
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.provider.Property
+import org.gradle.api.services.BuildService
+import org.gradle.api.services.BuildServiceParameters.None
+import org.gradle.api.tasks.Classpath
 import org.gradle.api.tasks.InputFiles
 import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.Optional
 import org.gradle.api.tasks.OutputFile
 import org.gradle.api.tasks.TaskAction
-import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 import java.io.File
 
 // Workaround for https://youtrack.jetbrains.com/issue/KT-38570
-open class DisableIncrementalCompilationTask : DefaultTask() {
-
-  @get:Internal lateinit var compileTask: KotlinCompile
-  @get:Internal var configurationName: String? = null
-    set(value) {
-      field = value
-      if (value != null) {
-        // Initialize the value at config time.
-        classpath = project
-            .configurations
-            .getByName(value)
-            .incoming
-            .artifacts("jar")
-            .files
-      }
-    }
-
-  @get:InputFiles lateinit var classpath: FileCollection
-    private set
+abstract class DisableIncrementalCompilationTask : DefaultTask() {
+
+  @Suppress("unused")
+  @get:Classpath
+  @get:InputFiles
+  abstract val pluginClasspath: ConfigurableFileCollection
 
   @Suppress("unused")
   @get:OutputFile @get:Optional
   val someFile = File(project.buildDir, "not-existing-file-because-gradle-needs-an-output")
```

Comment:
Can you use a RegularFileProperty here ? It would be more homogenous with the property below.

Reply:
I keep it for now. It's less code, less configuration and no unnecessary file is created on the file system that needs to be cached, packaged and uploaded.

Reply:
It's not so much of a change and it's the new standard way of doing things.